src/classes/ADDR_Cicero_Validator.cls

```diff
@@ -197,13 +197,15 @@ public with sharing class ADDR_Cicero_Validator implements ADDR_IValidator {
                             if (district.district_type == 'STATE_LOWER')
                                 addr.State_Lower_District__c = district.district_id;
                         }
-                    
-                        addr.MailingStreet__c = candidate.match_addr.split(',')[0].trim();
-                        addr.MailingCity__c = candidate.match_city;
-                        addr.MailingState__c = candidate.match_region;
-                        addr.MailingPostalCode__c = candidate.match_postal;
-                        if (!String.isBlank(lastFourPostalDigits)) {
-                            addr.MailingPostalCode__c += '-' + lastFourPostalDigits;
+
+                        if(!settings.Prevent_Address_Override__c){
```

Comment:
I think we should write the API response value if the field is blank, even if the setting is checked. So in your doesOverrideAddr() test, it wouldn't update the populated fields, but it would populate the missing City from the response.

Reply:
Got it, I can make changes for City and State for sure. Would there be a reasonable case where zip code or street would need to be populated if blank? There might be some random responses if that's the case. Currently taking a look at 2538 too.

Reply:
Good point, probably not Street, but definitely City, State, Zip, County, and Country.